tendrl/monitoring_integration/grafana/dashboard.py

```diff
@@ -30,15 +30,21 @@ def upload_default_dashboards():
     datasource.create()
     dashboards = dashboard_utils.get_all_dashboards()
     for dashboard_json in dashboards:
-        title.append(dashboard_json["uri"].split('/')[1])
-
-    for dashboard_json in NS.config.data["dashboards"]:
-        if dashboard_json in title:
-            msg = '\n' + "Dashboard " + str(dashboard_json) + \
-                  " already exists" + '\n'
+        dashboard = dashboard_json["uri"].split('/')[1]
+        title.append(dashboard)
+        # Deleting already existing dashboards if present
+        response = dashboard_utils.delete_dashboard(dashboard)
```

Comment:
We don't want to delete all dashboards in every sync. I think we need to find a better way of handling this situation. 

We need to see if there is an intersection/difference between the two files then only we need to update it. Otherwise, the will keep refreshing the dashboards and can cause a problem(while a user is at a particular dashboard) also it increases the number of calls on the server.

Reply:
@cloudbehl agreed, will see to it.